src/www/app/outline_server_repository/access_key_serialization.ts

```diff
@@ -29,40 +29,36 @@ export function staticKeyToShadowsocksSessionConfig(staticKey: string): Shadowso
       port: config.port.data,
       method: config.method.data,
       password: config.password.data,
-      prefix: config.extra['prefix'],
+      prefix: config.extra?.['prefix'],
     };
-  } catch (error) {
-    throw new errors.ServerAccessKeyInvalid(error.message || 'Failed to parse static access key.');
+  } catch (cause) {
+    throw new errors.ServerAccessKeyInvalid('Invalid static access key.', {cause});
   }
 }
 
 function parseShadowsocksSessionConfigJson(maybeJsonText: string): ShadowsocksSessionConfig | null {
-  let sessionConfig;
-  try {
-    const {method, password, server: host, server_port: port, prefix} = JSON.parse(maybeJsonText);
-
-    sessionConfig = {
-      method,
-      password,
-      host,
-      port,
-      prefix,
-    };
-  } catch (_) {
-    // It's not JSON, so return null.
-    return null;
-  }
+  const {method, password, server, server_port, prefix} = JSON.parse(maybeJsonText);
 
   // These are the mandatory keys.
-  for (const key of ['method', 'password', 'host', 'port']) {
-    if (sessionConfig && !sessionConfig[key]) {
-      throw new errors.ServerAccessKeyInvalid(
-        `Incomplete VPN information returned from dynamic access key: missing "${key}".`
-      );
+  const missingKeys = [];
+
+  for (const [key, value] of Object.entries({method, password, server, server_port})) {
+    if (value === undefined) {
```

Comment:
iirc `=== undefined` is not recommended?

```suggestion
    if (type value === `undefined`) {
```